chain-crypto/src/algorithms/vrf/vrf.rs

```diff
@@ -163,44 +165,44 @@ impl PublicKey {
         if input.len() != PUBLIC_SIZE {
             return Err(PublicKeyError::SizeInvalid);
         }
-        let ristretto = CompressedRistretto::from_slice(&input);
-        match ristretto.decompress() {
+        let group_element = GroupElement::from_bytes(input);
+        match group_element {
             None => Err(PublicKeyError::StructureInvalid),
-            Some(pk) => Ok(PublicKey(pk, ristretto)),
+            Some(pk) => Ok(PublicKey(pk.clone(), pk.to_bytes())),
         }
     }
 
     pub fn as_bytes(&self) -> &[u8] {
-        self.1.as_bytes()
+        &self.1
     }
 
     pub fn to_buffer(&self, output: &mut [u8]) {
         assert_eq!(output.len(), PUBLIC_SIZE);
-        output.copy_from_slice(self.0.compress().as_bytes())
+        output.copy_from_slice(&self.0.to_bytes())
     }
 }
 
 impl ProvenOutputSeed {
     /// Verify a proof for a given public key and a data slice
     pub fn verify(&self, public_key: &PublicKey, input: &[u8]) -> bool {
         let dleq = dleq::Dleq {
-            g1: &RISTRETTO_BASEPOINT_POINT,
+            g1: &GroupElement::generator(),
             h1: &public_key.0,
-            g2: &make_message_hash_point(input),
+            g2: &GroupElement::from_hash(&input),
             h2: &self.u.0,
         };
         dleq::verify(&dleq, &self.dleq_proof)
     }
 
     pub fn to_buffer(&self, output: &mut [u8]) {
         assert_eq!(output.len(), PROOF_SIZE);
-        output[0..32].copy_from_slice(self.u.0.compress().as_bytes());
+        output[0..32].copy_from_slice(&self.u.0.to_bytes());
```

Comment:
This will panic at runtime if `p256k1` feature is selected because `src` (`&self.u.0.to_bytes()`) is not the same length as `self` (`output[0..32]`)